lib/generateModelMaterialsCommon.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+var Cesium = require('cesium');
+var addExtensionsUsed = require('./addExtensionsUsed');
+
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+
+module.exports = generateModelMaterialsCommon;
+
+/**
+ * Converts materials to use the KHR_materials_common extension and deletes
+ * techniques, programs and shaders.
+ *
+ * The glTF asset must be initialized for the pipeline.
+ *
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @param {Object} [kmcOptions] KHR_materials_common options for material generation.
+ * @returns {Object} The glTF asset with generated normals.
+ *
+ * @see addPipelineExtras
+ * @see loadGltfUris
+ */
+function generateModelMaterialsCommon(gltf, kmcOptions) {
+    addExtensionsUsed(gltf, 'KHR_materials_common');
+    var materialsCommon;
+    var materials = gltf.materials;
+    var nodes = gltf.nodes;
+    var techniques = gltf.techniques;
+    var jointCountForMaterialId = getJointCountForMaterials(gltf);
+
+    var gltfExtensions = gltf.extensions;
+    if (!defined(gltfExtensions)) {
+        gltfExtensions = {};
+        gltf.extensions = gltfExtensions;
+    }
+    materialsCommon = {};
+    gltfExtensions.KHR_materials_common = materialsCommon;
+    var lights = {};
+    materialsCommon.lights = lights;
+
+    for (var materialId in materials) {
+        if (materials.hasOwnProperty(materialId)) {
+            var material = materials[materialId];
+            var technique = techniques[material.technique];
+            var techniqueParameters = technique.parameters;
+            if (defined(techniqueParameters.ambient) && !defined(lights.defaultAmbient)) {
+                lights.defaultAmbient = {
+                    ambient: {
+                        color: [1, 1, 1]
+                    },
+                    name: 'defaultAmbient',
+                    type: 'ambient'
+                };
+            }
```

Comment:
> I found an issue where Box doesn't quite have the right lighting post-generation. I think this is because there is no ambient light generated in the KHR_materials_common sample model, but I'm not sure when it's okay to not have it since it is generated for some of the others.

I'm not sure why an ambient light needs to be added. When Cesium or other engines encounter khr_materials_common with an ambient value wouldn't they just modify the shader?